sample.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+package ssf;
+
+message SSFTag {
+  string name = 1;
+  string value = 2;
+}
+
+message SSFTrace {
+  int64 trace_id = 1;
+  int64 id = 2;
+  int64 parent_id = 3;
+}
+
+message SSFSample {
+  enum Metric {
+      COUNTER = 0;
+      GAUGE = 1;
+      HISTOGRAM = 2;
+      SET = 3;
+      STATUS = 4;
+      EVENT = 5;
+      TRACE = 6;
+  }
+  enum Status {
+      OK = 0;
+      WARNING = 1;
+      CRITICAL = 2;
+      UNKNOWN = 3;
+  }
+  Metric metric = 1;
+  string name = 2;
+  int64 timestamp = 3;
+  string message = 4;
+  Status status = 5;
+  // At the moment, this Value refers specifically to a Duration
+  // TODO change this to a separate field
+  int64 value = 6;
+  float sample_rate = 7;
+  repeated SSFTag tags = 8;
+  string unit = 9;
+  SSFTrace trace = 10;
+  string resource = 11;
+  string service = 12;
```

Comment:
the proto structure here is reasonably complex. as i recall when we were pairing on this, it wasn't immediately obvious what fields meant what and with what values. can we comment the shit out of this?